Documentation/gettingstarted/servicemesh/envoy-custom-listener.rst

```diff
@@ -0,0 +1,70 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _gs_envoy_custom_listener:
+
+*******************
+L7 Path Translation
+*******************
+
+This example replicates the Prometheus metrics listener which is
+already available via Cilium Agent ``--proxy-prometheus-port`` command
+line option. So the point of this example is not to add new
+functionality, but to show how a feature that previously required
+Cilium Agent code changes can be implemented with the new Cilium Envoy
+Config CRD.
+
+Apply Example CRD
+=================
+
+This example adds a new Envoy listener ``envoy-prometheus-metrics-listener``
+on the standards Prometheus port (e.g. ``9090``) to each Cilium node, translating
+the default Prometheus metrics path ``/metrics`` to Envoy's Prometheus metrics path
+``/stats/prometheus``.
+
+Apply this Cilium Envoy Config CRD:
+
+.. parsed-literal::
+
+    $ kubectl apply -f \ |SCM_WEB|\/examples/kubernetes/servicemesh/envoy/envoy-prometheus-metrics-listener.yaml
+
+This version of the CEC CRD is Cluster-scoped, (i.e., not namespaced),
+so the name needs to be unique in the cluster, unless you want to
+replace a CRD with a new one.
+
+.. Note::
+
+    Note that these Envoy resources are not validated by k8s at all, so
+    any errors in the Envoy resources will only be seen by the Cilium
+    Agent observing these CRDs. This means that ``kubectl apply`` will
+    report success, while parsing and/or installing the resources to the
+    node-local Envoy instance may have failed. Currently the only way of
+    verifying this is by observing Cilium Agent logs for errors and
+    warnings.
```

Comment:
Probably these warnings should be in a dedicated file and included from the `envoy-traffic-management.rst` file as well, where that file is configuring CEC.

Reply:
Eh, then again if `envoy-traffic-management.rst` is not getting the user to configure anything then maybe this one isn't as important to include over on that page since it's less likely something gets misconfigured.